File: 0345-reverse-vowels-of-a-string/solution.c
#include<string.h>
char* reverseVowels(char* a) {
    int n=strlen(a);
    char s[100000];
    int count=0;
    for(int i=n-1;i>=0;i--){
        if( a[i]=='a' || a[i]=='i' || a[i]=='e' || a[i]=='o' || a[i]=='u' ||a[i]=='A' || a[i]=='I' || a[i]=='E' || a[i]=='O' || a[i]=='U')
        {
            s[count++]=a[i];
           
        }
    }
    int c=0;
    for(int i=0;i<n;i++){
        if( a[i]=='a' || a[i]=='i' || a[i]=='e' || a[i]=='o' || a[i]=='u'||  a[i]=='A' || a[i]=='I' || a[i]=='E' || a[i]=='O' || a[i]=='U'){
            a[i]=s[c++];
            
        }
    }
    return a;
}

File: 0345-reverse-vowels-of-a-string/solution.java
class Solution {
    public String reverseVowels(String s) {
        int n=s.length();
        int left=0;
        int right=n-1;
        Set<Character> vowels = new HashSet<>();
        for(char c: "aeiouAEIOU".toCharArray()){
            vowels.add(c);
        }
        char[] chars = s.toCharArray();
        while(left<right){
            while(left<right && !vowels.contains(chars[left])){
                left++;
            }
            System.out.println(left);
            while(left<right && !vowels.contains(chars[right])){
                right--;
            }
            System.out.println(right);
            char temp = chars[left];
            chars[left] = chars[right];
            chars[right]=temp;

            left++;
            right--;

        }
        return new String(chars);
    }
}
